src/lib/components/VideoCard/VideoCard.svelte

```diff
@@ -1,78 +1,151 @@
+  const videoCardPadding = "3rem";
+  const videoCardFlexGap = "1.5rem";
+  const descriptionWidth = "200px";
+
+  const mobileSizes = [
+    "(max-width: 640px) calc(100% - (${bodyPadding} + ${videoCardPadding}))",
+    `(max-width: 1023px) calc(100% - (${bodyPadding} + ${videoCardPadding} + ${videoCardFlexGap} + ${descriptionWidth}))`,
+  ];
+
+  const sizesBySizeType: Record<SizeType, string> = {
+    "hero-full-width": [...mobileSizes, "688px"].join(", "),
+    "hero-col-9": [...mobileSizes, "420px"].join(", "),
+  };
+
+  let showYoutubeEmbed = !thumbnails;
+  let lastVideoURL = url;
   $: {
-    const videoUrl = new URL(url);
-    if (videoUrl.hostname.includes("youtube.com")) {
-      youtubeVideoId = videoUrl.searchParams.get("v");
-    } else if (videoUrl.hostname.includes("youtu.be")) {
-      youtubeVideoId = videoUrl.pathname.split("/")[1];
-    } else {
-      youtubeVideoId = null;
-      console.warn("Unsupported video type; could not embed video.");
-    }
+    if (lastVideoURL !== url) showYoutubeEmbed = false;
+    lastVideoURL = url;
   }
 
-  $: title = customTitle;
-  $: description = customDescription;
-
-  afterUpdate(async () => {
-    if (youtubeVideoId && (!customTitle || !customDescription)) {
-      const snippet = await getYoutubeVideoData(youtubeVideoId);
-      if (!customTitle) {
-        title = snippet.title;
-      }
-      if (!customDescription) {
-        // Only use the first paragraph instead of rendering with the full
-        //   description to avoid trimming at a newline (leaving an orphaned
-        //   ellipsis) and to hopefully avoid displaying URLs and hashtags.
-        description = snippet.description.split("\n")[0];
-      }
-    }
-  });
+  const handleKeyDown = (e: KeyboardEvent) => {
+    if (!(e.key === "Enter" || e.key === " ")) return;
+    if (showYoutubeEmbed) return;
+    e.preventDefault();
+    showYoutubeEmbed = true;
+  };
 </script>
 
-{#if youtubeVideoId}
+{#if youtubeVideoID}
   <div class="ldaf-video-card ldaf-video-card--{variation} {className}">
     <div class="ldaf-video-container">
-      <iframe
-        class="ldaf-video-embed"
-        src={`https://www.youtube-nocookie.com/embed/${youtubeVideoId}?rel=0`}
-        title="Embedded YouTube video player"
-        frameborder="0"
-        allow="encrypted-media;"
-        allowfullscreen
-      />
+      <noscript>
+        <iframe
+          class="ldaf-video-embed"
+          src={`https://www.youtube-nocookie.com/embed/${youtubeVideoID}?rel=0`}
+          title="Embedded YouTube video player"
+          frameborder="0"
+          allow="encrypted-media;"
+          allowfullscreen
+        />
+      </noscript>
+      {#if showYoutubeEmbed}
+        <iframe
+          class="ldaf-video-embed"
+          src={`https://www.youtube-nocookie.com/embed/${youtubeVideoID}?rel=0&autoplay=1`}
+          title="Embedded YouTube video player"
+          frameborder="0"
+          allow="encrypted-media;"
```

Comment:
In order to get autoplay working, you'll also need to edit this line:
```suggestion
          allow="autoplay;encrypted-media;"
```